pkg/operation/garden/garden.go

```diff
@@ -55,6 +55,10 @@ func (b *Builder) WithProject(project *gardencorev1beta1.Project) *Builder {
 func (b *Builder) WithProjectFrom(reader client.Reader, namespace string) *Builder {
 	b.projectFunc = func(ctx context.Context) (*gardencorev1beta1.Project, error) {
 		project, _, err := gutil.ProjectAndNamespaceFromReader(ctx, reader, namespace)
+		if project == nil {
+			return nil, fmt.Errorf("cannot find Project for namespace '%s'", namespace)
+		}
```

Comment:
I think we first want to check the err:

```suggestion
		if err != nil {
			return nil, err
		}
		if project == nil {
			return nil, fmt.Errorf("cannot find Project for namespace '%s'", namespace)
		}
```

For example the err can actually be "namespace not found" or "project not found" or any other but we would always kind of ignore this error and return "cannot find Project for namespace" err.